Products/Zuul/infos/devicemanagement.py

```diff
@@ -38,8 +40,26 @@ def startTime(self):
         return self._object.niceStartDateTime()
 
     def updateWindow(self, p):
+        startDateTime = p.get('startDateTime', None)
+        if not startDateTime:
+            startDate = p.get('startDate',
+                datetime.datetime.now().strftime('%m/%d/%Y'))
+            startHours = p.get('startHours', '00')
+            startMinutes = p.get('startMinutes', '00')
+            startSeconds = p.get('startSeconds', '00')
+            startDateTimeString = '{} {}:{}:{}'.format(
+            startDate, startHours, startMinutes, startSeconds)
+            dateTime = datetime.datetime.strptime(startDateTimeString,
+                "%m/%d/%Y %H:%M:%S")
+            startDateTime = time.mktime(dateTime.timetuple())
+        duration = int(p['durationMinutes']) or int(p['durationHours']) \
+            or int(p['durationDays'])
+        if not duration:
+            p['durationHours'] = 1
+        if p['repeat'] not in self._object.REPEAT:
+            p['repeat'] = self._object.REPEAT[0]
         self._object.manage_editMaintenanceWindow( 
-                                     startDateTime=p['startDateTime'],
+                                     startDateTime=startDateTime,
                                      durationDays=p['durationDays'],
```

Comment:
Seems like this method is duplicating some default value picking logic defined in the manage_editMaintenanceWindow param and method.  Specifically the logic around startDateTime and startHours etc.... Those values should be passed in only if they are present in the the `p` param.

Why is it default to 1 hour duration if the duration not set?

Seems like an exception should be thrown if an invalid `repeat` param is sent.